src/edtr-io/plugins/h5p.tsx

```diff
@@ -22,32 +38,146 @@ export const H5pPlugin: EditorPlugin<H5pPluginState> = {
+              vier Inhaltstypen:
+              <ul>
+                <li>Fill in the Blanks</li>
+                <li>Drag the Words</li>
+                <li>Find the Hotspot</li>
+                <li>Drag and Drop</li>
+              </ul>
+            </li>
+            <li>
+              Erstelle deinen Inhalt, speichere ihn und klicke dann auf
+              &quot;Inhalt bereitstellen&quot;.
+            </li>
+            <li>Füge den Bereitstellungslink hier ein:</li>
+          </ul>
+        </div>
+        <div>
+          <EditorInlineSettings>
+            <EditorInput
+              label="URL zu Lumi-Bereitstellung"
+              placeholder="https://app.lumi.education/run/P46cdL"
+              value={state.value}
+              onInput={(e: React.ChangeEvent<HTMLInputElement>) => {
+                //console.log('on input', e.target.value)
+                if (
+                  e.target.value
+                    .toLowerCase()
+                    .startsWith('https://app.lumi.education/run/')
+                ) {
+                  setError('')
+                } else {
+                  setError(
+                    'Die URL muss mit https://app.lumi.education/run/ beginnen.'
+                  )
+                }
+                setUrl(e.target.value)
+                state.set(e.target.value)
+              }}
+              inputWidth="70%"
+              width="100%"
+              ref={props.autofocusRef}
+            />
+          </EditorInlineSettings>
+        </div>
+        {error && <p className="text-red-500 mt-2 mx-side">{error}</p>}
+        <p>
+          <button
+            className="mt-2 serlo-button bg-brandgreen-300 disabled:bg-gray-300 disabled:cursor-default"
+            disabled={!!(!url || error || mode == 'loading')}
+            onClick={() => {
+              setMode('loading')
+              void (async () => {
+                const match = /https:\/\/app\.lumi\.education\/run\/(.+)/i.exec(
+                  url
+                )
+                try {
+                  const id = match ? match[1] : ''
+                  const res = await fetch(
+                    'https://app.lumi.education/api/v1/run/' + id
+                  )
+                  const json = (await res.json()) as {
+                    dependencies: { machineName: string }[]
+                  }
+                  if (
+                    !json.dependencies.every((dep) =>
+                      h5pLibraryWhitelist.includes(dep.machineName)
+                    )
+                  ) {
+                    setError(
+                      'Unerlaubte Plugintypen - nutze bitte nur die vier genannten Inhaltstypen'
+                    )
+                    setMode('edit')
+                  }
+                  //console.log(json)
+                  props.state.set(url)
+                  setMode('preview')
+                } catch (e) {
+                  alert(e)
```

Comment:
could this also be handled with setError?

Reply:
I will investigate this